middleware/auth-utils/grant-manager.js

```diff
@@ -100,6 +102,14 @@ GrantManager.prototype.obtainFromCode = function obtainFromCode (request, code,
   const handler = createHandler(this);
   const options = postOptions(this);
 
+  if (this.authServerHost) {
+    options.hostname = this.authServerHost;
+  }
+
+  if (this.authServerPort) {
+    options.port = this.authServerPort;
+  }
```

Comment:
I'm a little unclear if these settings should only be set here, or if they should additionally be set inside of `obtainDirectly`, `obtainFromClientCredentials`, etc. If so, then the change should probably occur inside of the `postOptions` function, and the code would instead be the follow. I'm just not sure what the ramifications of that would be for other credential methods.

```
if (manager.authServerHost) {
  opts.hostname = manager.authServerHost;
}

if (manager.authServerPort) {
  opts.port = manager.authServerPort;
}
```